src/App.tsx

```diff
@@ -1,203 +1,102 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { FilterType } from './types/FilterType';
+import { getFilteredTodos } from './utils/getFilteredTodo';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(FilterType.all);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(
+    ErrorType.default,
+  );
+
+  const allIsCompleted = todos.every(todo => todo.completed);
+
+  const handleFilterChange = useCallback((filterBy: FilterType) => {
+    setFilter(filterBy);
+  }, []);
+
+  const filteredTodos = useMemo(
+    () => getFilteredTodos(todos, filter),
+    [todos, filter],
+  );
+
+  const handleRemoveError = () => {
+    setErrorMessage(ErrorType.default);
+  };
+
+  const handleErrorMessage = useCallback((error: ErrorType) => {
+    setErrorMessage(error);
+
+    setTimeout(handleRemoveError, 3000);
+  }, []);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => handleErrorMessage(ErrorType.load));
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.